src/stories/components/Menu/styles.ts

```diff
@@ -0,0 +1,212 @@
+import React from 'react';
+
+import { Menu } from '@mui/icons-material';
+import CloseIcon from '@mui/icons-material/Close';
+import { ListItem, ListItemText } from '@mui/material';
+import Button from '@mui/material/Button';
+import Divider from '@mui/material/Divider';
+import Grid from '@mui/material/Grid';
+import List from '@mui/material/List';
+import { styled } from '@mui/material/styles';
+import SwipeableDrawer from '@mui/material/SwipeableDrawer';
+import { fontSizes } from 'theme/fontSizes.theme';
+import { spacing } from 'theme/spacing';
+
+export const GridHeader = styled(Grid)`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+
+  &.data-user {
+    padding-left: ${spacing.S16};
+    gap: ${spacing.S16};
+  }
+`;
+
+export const FixedMenu = styled('div')`
+  position: fixed;
+  width: 15rem;
+  height: 100%;
+  left: 0;
+  top: 0;
+  z-index: 1000;
+  background: #000;
+
+  & .MuiDrawer-paper {
+    background: #000;
+    box-shadow: none;
+    overflow: hidden;
+  }
+
+  backdrop-filter: blur(1.9375rem);
+  background: hsla(0, 0%, 100%, 0.65);
+`;
+
+export const MenuButton = styled(Button)`
+  color: #000;
+  border-radius: 50%;
+  border-color: black;
+  height: 1.875rem;
+  min-width: 0;
+  width: 1.75rem;
+
+  &:hover {
+    background: none;
+    border-color: unset;
+  }
+`;
+
+export const MenuIcon = styled(Menu)`
+  font-size: ${fontSizes.F16};
+`;
+
+export const GridSeeAll = styled(Grid)`
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+`;
+
+export const MenuDrawer = styled(SwipeableDrawer)`
+  & .MuiDrawer-paper {
+    background: transparent;
+    box-shadow: none;
+
+    overflow: hidden;
+  }
+`;
+
+export const CloseDrawerButton = styled(Button)`
+  color: #000;
+  border-radius: 50%;
+  border-color: #000;
+  height: 1.875rem;
+  min-width: 0;
+  width: 1.75rem;
+
+  &:hover {
+    background: none;
+    border-color: unset;
+  }
+`;
+
+export const DrawerGrid = styled(Grid)`
+  height: 100%;
+
+  ${({ theme }) => `
+  ${theme.breakpoints.up('xs')} {
+     width: 18.125rem;
+   }
+   ${theme.breakpoints.up('md')} {
+    width: 21.25rem;
+  }
+  `}
+
+  backdrop-filter: blur(1.9375rem);
+  background: hsla(0, 0%, 100%, 0.65);
+`;
+
+export const LogoWrapper = styled(Grid, {
+  shouldForwardProp: (prop) => !['logo'].includes(prop as string)
+})<{ logo?: React.ReactNode }>`
+  width: ${(props) => (props.logo ? 'auto' : '9.375rem')};
+  height: ${(props) => (props.logo ? 'auto' : '1.875rem')};
+`;
```

Comment:
A prop logo neste caso não precisa ser um React.ReactNode, pode ser simplesmente um booleano (pois você só está interessada em saber se tem logo ou não).

```suggestion
export const LogoWrapper = styled(Grid, {
  shouldForwardProp: (prop) => !['hasLogo'].includes(prop as boolean)
})<{ hasLogo?: boolean }>`
  width: ${(props) => (props.hasLogo ? 'auto' : '9.375rem')};
  height: ${(props) => (props.hasLogo ? 'auto' : '1.875rem')};
`;
```